cfg/validate.go

```diff
@@ -232,5 +241,9 @@ func ValidateConfig(v isSet, config *Config) error {
 		return fmt.Errorf("error parsing metrics config: %w", err)
 	}
 
+	if err = isValidParallelDownloadConfig(config); err != nil {
```

Comment:
Shouldn't we check emptiness of cacheDir outside and pass only FileCacheConfig to validate parallelDownload parameters?

Reply:
I did thought about this. But I found that, it is cleaner to pass top level object and encapsulate details inside a function. Since there is one part of this config is already outside of file cache (not ideal!) there is a possibility of something else coming at the top level hence I found this approach better than polluting this function.